iPetCare/Application/Services/UserService.cs

```diff
@@ -83,6 +83,105 @@ public async Task<ServiceResponse<GetAllUsersDtoResponse>> GetAllAsync()
             return new ServiceResponse<GetAllUsersDtoResponse>(HttpStatusCode.OK, dto);
         }
 
+        public async Task<ServiceResponse<EditProfileDtoResponse>> EditProfileAsync(EditProfileDtoRequest dto)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<EditProfileDtoResponse>(HttpStatusCode.Unauthorized);
+
+            if (CurrentlyLoggedUser.Role == Role.Owner)            
+                return await EditOwnerProfileAsync(dto);            
+
+            if (CurrentlyLoggedUser.Role == Role.Vet)
+                return await EditVetProfileAsync(dto);
+
+            if (CurrentlyLoggedUser.Role == Role.Administrator)
+                return await EditAdminProfileAsync(dto);
+
+            return new ServiceResponse<EditProfileDtoResponse>(HttpStatusCode.BadRequest);
+        }
+
+        private async Task<ServiceResponse<EditProfileDtoResponse>> EditAdminProfileAsync(EditProfileDtoRequest dto)
+        {
+            if (await Context.Users.Where(x => x.Email == dto.Email && CurrentlyLoggedUser.Email != dto.Email).AnyAsync())
+                return new ServiceResponse<EditProfileDtoResponse>(HttpStatusCode.BadRequest, "Email jest zajęty");
+
+            if (await Context.Users.Where(x => x.UserName == dto.UserName && CurrentlyLoggedUser.UserName != dto.UserName).AnyAsync())
```

Comment:
Czy testowałaś jak to się zachowa przy zmianie nicku?

Chodzi o to, że CurrentUser ustalany jest na podstawie tego, co ma ASP NET, czyli currentPrincipala. Ten principal w sobie ma User.Name. Ten Name to username użytkownika. Jeśli do tej pory miał username np. "abc" a zmienił sobie na "qwe", to czy w kolejnych requestach HttpContext.User.Identity.Name będzie miał abc czy qwe? Wydaje mi się, że w przypadku zmiany loginu czy emaila warto by było wylogować użytkownika, aby mieć pewność.

Natomiast to proszę skonsultuj jeszcze z Pawłem.

Reply:
O tym już wspomniałam na konwersacji :D